src/events/messageCreate/checkAvailability.js

```diff
@@ -0,0 +1,207 @@
+                        'type':        'array',
+                        'description': 'An array of booleans representing the days the user is available. Only monday through friday.',
+                        'items':       {
+                            'monday': {
+                                'type':        'bool',
+                                'description': 'If monday is available or not.',
+                            },
+                            'tuesday': {
+                                'type':        'bool',
+                                'description': 'If tuesday is available or not.',
+                            },
+                            'wednesday': {
+                                'type':        'bool',
+                                'description': 'If wednesday is available or not.',
+                            },
+                            'thursday': {
+                                'type':        'bool',
+                                'description': 'If thursday is available or not.',
+                            },
+                            'friday': {
+                                'type':        'bool',
+                                'description': 'If friday is available or not.',
+                            }
+                        }
+
+                    }
+                },
+                'required': ['from', 'to', 'days'],
+            },
+        },
+    },
+    {
+        'type':     'function',
+        'function': {
+            'name':        'unableToParse',
+            'description': 'If the user inputs content that cannot be used in the other functions.',
+            'parameters':  {},
+        },
+    },
+];
+
+/**
+ * Given a function object from open ai and a message, attempt to parse 
+ * @param {DiscordJS Message} message 
+ * @param {OpenAI Tool} calledFunction 
+ */
+const parseResults = (message, calledFunction) =>{
+    const action = {
+        'rememberUnavailability': () => rememberUnavailability(message, JSON.parse(calledFunction.arguments)),
+        'rememberAvailability':   () => rememberAvailability(message, JSON.parse(calledFunction.arguments)),
+        'unableToParse':          () => unableToParse(message),
+    };
+    action[calledFunction.name]();
+};
+
+// checks the message sent in the channel and sends it to openai to parse, then if possible saves the availability data
+module.exports = async (client, message) => {
+    try {
+        const availabilityChannel = await getAvailabilityChannel();
+
+        // only send a message if it's not from a bot and it's from the available channel
+        if (message.author.bot || !availabilityChannel || message.channelId !== availabilityChannel.id) {
+            return;
+        }
+
+        const date = new Date(message.createdTimestamp).toLocaleDateString();
+        const response = await openAiClient.chat.completions.create({
+            model:    'gpt-3.5-turbo',
+            messages: [
```

Comment:
Probably also want to include a system prompt, basically telling the model about itself and it's goals.